### ISSUE TITLE
Allow input from user on actual matches

### ISSUE THREAD

**Issue by thibautjombart** [Hackout3/epimatch#10]
Not sure if this is sci-fi, as it sounds like machine learning, but the idea would be for the user to tick boxes indicating actual matches, refine the search, etc.

**Comment by kplaney** [comment]
Thanks for the idea! Haha totally fine if it sounds Sci Fi -  it would be interesting to think about how to do this.

I need to update the README (work got the best of me right after the Hackathon) but I volunteered to maintain the repo.

**Comment by zkamvar** [comment]
👽 

Since the underlying structure is distance matrices for each category, I wonder if this could take the form of modification of the weights. I'd imagine it would take some refactoring to be able to allow for updates (creating new objects and whatnot, but it's worth investigating!).

**Comment by thibautjombart** [comment]
Yeah, I think you could treat that as an optimization problem, i.e. find the vector of weights which maximizes the fit to user's classification of matches. One question will be on the choice of the quantity optimized; I can see:
- sensitivity / True Positive Rate: proportion of actual (user-defined) matches correctly identified
- specificity / True Negative Rate: proportion of non-match classified as non-match
- F-score: a mixture of the two (see https://en.wikipedia.org/wiki/Sensitivity_and_specificity)

That would be a fairly easy problem I think (param space is small), but worth a bit of research (and probably a paper). Happy to help with the stats bits / proof of concept.

What d'yall think?
